# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/modules/requested-modules.js
# Commit: d9fa28df9787
# Full Hash: d9fa28df9787b4d01135df632c987f20e830b9d6
# Author: Jonatan Klemets <jonatan.r.klemets@gmail.com>
# Date: 2021-12-02 09:42:49
# Regressor Bug: 1736060
# File Overlap Count: 1
# Description:
#   Bug 1736060 - Part 5: Implement import assertions for static imports. r=mgaudet,arai
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D126045
# ==============================================================================

diff -r 5b9622ccd0d2 -r d9fa28df9787 js/src/jit-test/tests/modules/requested-modules.js
--- a/js/src/jit-test/tests/modules/requested-modules.js	Wed Dec 01 18:26:09 2021 +0000
+++ b/js/src/jit-test/tests/modules/requested-modules.js	Wed Dec 01 18:26:10 2021 +0000
@@ -5,26 +5,79 @@
     var actual = module.requestedModules;
     assertEq(actual.length, expected.length);
     for (var i = 0; i < actual.length; i++) {
-        assertEq(actual[i].moduleRequest.specifier, expected[i]);
+        assertEq(actual[i].moduleRequest.specifier, expected[i].specifier);
+        if(expected[i].assertions === null) {
+            assertEq(actual[i].moduleRequest.assertions, null);
+        }
+        else {
+            var expectedAssertions = expected[i].assertions;
+            var actualAssertions = actual[i].moduleRequest.assertions;
+            assertEq(actualAssertions.length, expectedAssertions.length);
+            for (var j = 0; j < expectedAssertions.length; j++) {
+                assertEq(expectedAssertions[j].type, actualAssertions[j].type);
+            }
+        }
     }
 }
 
 testRequestedModules("", []);
 
-testRequestedModules("import a from 'foo'",
-                     ['foo']);
+testRequestedModules("import a from 'foo'", [
+    { specifier: 'foo', assertions: null }
+]);
+
+testRequestedModules("import a from 'foo'; import b from 'bar'", [
+    { specifier: 'foo', assertions: null },
+    { specifier: 'bar', assertions: null }
+]);
+
+testRequestedModules("import a from 'foo'; import b from 'bar'; import c from 'foo'", [
+    { specifier: 'foo', assertions: null },
+    { specifier: 'bar', assertions: null }
+]);
 
-testRequestedModules("import a from 'foo'; import b from 'bar'",
-                     ['foo', 'bar']);
+testRequestedModules("export {} from 'foo'", [
+    { specifier: 'foo', assertions: null }
+]);
 
-testRequestedModules("import a from 'foo'; import b from 'bar'; import c from 'foo'",
-                     ['foo', 'bar']);
+testRequestedModules("export * from 'bar'",[
+    { specifier: 'bar', assertions: null }
+]);
+
+testRequestedModules("import a from 'foo'; export {} from 'bar'; export * from 'baz'", [
+    { specifier: 'foo', assertions: null },
+    { specifier: 'bar', assertions: null },
+    { specifier: 'baz', assertions: null }
+]);
 
-testRequestedModules("export {} from 'foo'",
-                     ['foo']);
+if(getRealmConfiguration()['importAssertions']) {
+    testRequestedModules("import a from 'foo' assert {}", [
+        { specifier: 'foo', assertions: null },
+    ]);
+
+    testRequestedModules("import a from 'foo' assert { type: 'js'}", [
+        { specifier: 'foo', assertions: [ { type: 'js' } ] },
+    ]);
+
+    testRequestedModules("import a from 'foo' assert { unsupported: 'test'}", [
+        { specifier: 'foo', assertions: null },
+    ]);
 
-testRequestedModules("export * from 'bar'",
-                     ['bar']);
+    testRequestedModules("import a from 'foo' assert { unsupported: 'test', type: 'js', foo: 'bar' }", [
+        { specifier: 'foo', assertions: [ { type: 'js' } ] },
+    ]);
 
-testRequestedModules("import a from 'foo'; export {} from 'bar'; export * from 'baz'",
-                     ['foo', 'bar', 'baz']);
+    testRequestedModules("import a from 'foo' assert { type: 'js1'}; export {} from 'bar' assert { type: 'js2'}; export * from 'baz' assert { type: 'js3'}", [
+        { specifier: 'foo', assertions: [ { type: 'js1' } ] },
+        { specifier: 'bar', assertions: [ { type: 'js2' } ] },
+        { specifier: 'baz', assertions: [ { type: 'js3' } ] }
+    ]);
+
+    testRequestedModules("export {} from 'foo' assert { type: 'js'}", [
+        { specifier: 'foo', assertions:  [ { type: 'js' } ] }
+    ]);
+
+    testRequestedModules("export * from 'bar' assert { type: 'json'}",[
+        { specifier: 'bar', assertions:  [ { type: 'json' } ] }
+    ]);
+}
\ No newline at end of file